### DISCUSSION TITLE
2022년 개발 방향

### DISCUSSION THREAD

**Discussion by woonjjang** [team-durumi/kctu-photo#18] [Ideas]
2022년 개발하자는 부분에 대해서 우리가 어떻게 방향을 잡을지는 논의를 좀 더 해보시죠 !

계약 : https://github.com/team-durumi/ops/issues/88

## 개발방향 
1. Hugo & markdown 스크립트를 더 밀고 가 본다. 
-  파일의 변경사항을 bash 영역에서 계속 체크하는 방식으로 ?
- 지금보다는 좀 더 자동화
- md file 및 양이 6배 정도 더 많아지면, build도 1시간이 넘을텐데

2. Hugo & Svelte(items 목록 조회 / 세부) &  검색을 위한 DB
3. 많이 해 본 Application 개발 영영으로 회귀(Drupal, Rails) - 서버, DB

## 고객이 해결하고 싶어하는 것 중 critical 한 부분
1. 자료(사진 폴더) 업데이트 / 태그, 사진 제목같은 메타 수정
2. 검색엔진 붙이자.
3. 2022년에는 NAS쪽 편집자들은 늘어날 수 있음(KCTU-Rails-상담 사용자와 비슷한 조직구조)

**Comment by mozodev** [comment]
### AS-IS

- [1] 파일시스템 => 웹사이트
- [2] 파일시스템 + 태그 메타 csv => 웹사이트
- [3] 파일시스템 + 태그 메타 csv + 컴포넌트용 S3 => 웹사이트

이런식으로 소스로부터 빌드하는 경우에는 원본 소스 NAS 상태 diff 하는게 사실 상 의미가 없음
전체 빌드할 때 현재 소스가 그대로 웹사이트로 빌드됨.

### 문제: 고객도 관리가 단순하지 않음.

고객도 원본 NAS가 아니라 별도로 공개용 디렉토리에 정리하니까 헤깔리고 
거기에 파일시스템에 기록할 수 없는 n개의 태그정보까지 스프레드시트로 분리하면서 관리포인트가 2개로 증가.
이렇게 되면 처음 의도했던 `폴더 => 웹사이트` 격렬한 단순함이라는 장점은 이미 날아간 상태.

### 문제: 디렉토리명 변경 등 파일시스템 구조 변경 시 S3 동기화?

디렉토리명을 바꾸거나 하면 서비스용 S3에도 반영을 해줘야... 여기서 관리포인트 +1
재입수 스크립트를 돌리면 휴고 콘텐츠는 생성되겠.... 고객의 실수나 잘못된 데이터 구조가 있으면 이것도 빌드 실패.. ㅎ
반복해서 이야기했지만 파일과 아이템의 식별자가 없기 때문에 추적할 수 없으므로 기존 구조를 그대로 사용해야함.
변경된 내용을 알아야 S3를 수정할 수 있음. 
변경내역이 많을 경우 사람이 눈으로 하면 고통스럽고,
자동화하려면 스크립트로 해야 하는데 잘 모르겠...

**Reply by mozodev** [reply]
생각해보니 S3 동기화 이슈는 모델 메타와 파일이 따로 놀아서 생기는... 그렇다면 같이 하면? 
휴고 마크다운과 이미지를 분리하지 않고 동일한 디렉토리에서 빌드 시 설정 분기!

config.yml
- 마크다운 파일 무시 (이미지만 빌드) html이 아니라 이미지 파일들을 빌드하면 휴고 파이프로 이미지 리사이즈도 가능. S3 마운트 포인트로 결과물 출력.
- 이미지 파일 무시 (마크다운만 빌드) 지금 netlify 빌드와 동일.

**Reply by mozodev** [reply]
어쨌든 아이템 벌크 수정하려면 관리자가 필요하고 이건 netlify CMS로 대응이 안됨.
어쨌든 디비 만들고 관리자 화면이 필요. 디비에 수정하면 다시 hugo로 출력해서 기존 pipe를 태워야함.
글로 적는데도 아 정말 많은 과정이... 그런데 이런 공정으로부터 얻을 수 있는 장점은 단순한 스태틱 웹사이트 유지!
